### PR TITLE
OpenSSL 3 support

### PR DESCRIPTION
**Pull request by quitschbo**
This reverts commit 6b38096d06e3c835bc5d117dd7b096872ece83a3. The CML layer is now compatible with OpenSSL 3. Thus, we can remove this backport and switch to the upstream OpenSSL recipe.

Needs: https://github.com/gyroidos/cml/pull/392, 
https://github.com/gyroidos/meta-trustx-intel/pull/28,
https://github.com/gyroidos/gyroidos_build/pull/108

### CODE REVIEW

File: recipes-tss/openssl-tpm2-engine/openssl-tpm2-engine_4.0.1.bb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+SRC_URI[sha256sum] = "42fcecaf40945ae9603b93bc1e63a48b62ac1b25f9b4375c3b67dac9d5bf1ea7"
+
+inherit autotools pkgconfig
+
+do_configure:prepend() {
+	touch ${S}/NEWS
+	touch ${S}/AUTHORS
+	touch ${S}/ChangeLog
+	cp ${S}/LICENSE ${S}/COPYING
+}
```
Comment (jwsbck) [inline recipes-tss/openssl-tpm2-engine/openssl-tpm2-engine_4.0.1.bb:39]: Just out of interest: why do we place these empty files into the source directory? While this shouldn't hurt, i wonder what's the purpose and where this is needed. Also - if required - would it make more sense to place them in the ${B} directory?

Comment (quitschbo) [reply recipes-tss/openssl-tpm2-engine/openssl-tpm2-engine_4.0.1.bb:39]: This is for the sanity check. This was copied from the previous version of the recipe

**Review by jwsbck — APPROVED** [review]

**Comment by quitschbo** [comment]
Hmm the kernel do_deploy() after initial ram-disk bundling produces a Segfault. with switching to the current openssl3 recipe. So do not merge, yet.

**Comment by quitschbo** [comment]
sbsign was not supporting openssl 3. The PR https://github.com/gyroidos/meta-trustx-intel/pull/28 fixes that by using more recent version of sbsigntool

**Pull request merged**